File: swift/Sources/HeddleActor/Subjects.swift
public enum HeddleSubjects {
    public static let incomingTasks = "heddle.tasks.incoming"
    public static let deadLetters = "heddle.tasks.dead_letter"
    public static let incomingGoals = "heddle.goals.incoming"
    public static let controlReload = "heddle.control.reload"

    public static func workerTasks(workerType: String, tier: String) -> String {
        "heddle.tasks.\(workerType).\(tier)"
    }

    public static func results(parentTaskId: String?) -> String {
        "heddle.results.\(parentTaskId ?? "default")"
    }

    public static func processorQueueGroup(workerType: String) -> String {
        "processors-\(workerType)"
    }

    public static func llmWorkerQueueGroup(workerType: String) -> String {
        "workers-\(workerType)"
    }
}

File: dotnet/src/Heddle.Sdk/Subjects.cs
namespace Heddle.Sdk;

/// <summary>
/// Heddle NATS subject and queue-group conventions.
/// </summary>
public static class HeddleSubjects
{
    public const string IncomingTasks = "heddle.tasks.incoming";
    public const string DeadLetters = "heddle.tasks.dead_letter";
    public const string IncomingGoals = "heddle.goals.incoming";
    public const string ControlReload = "heddle.control.reload";

    public static string WorkerTasks(string workerType, string tier)
    {
        return $"heddle.tasks.{workerType}.{tier}";
    }

    public static string Results(string? parentTaskId)
    {
        return $"heddle.results.{parentTaskId ?? "default"}";
    }

    public static string ProcessorQueueGroup(string workerType)
    {
        return $"processors-{workerType}";
    }

    public static string LlmWorkerQueueGroup(string workerType)
    {
        return $"workers-{workerType}";
    }
}

